clients/mongoStoreClient.go

```diff
@@ -45,6 +47,21 @@ func NewMongoStoreClient(config *mongo.Config) *MongoStoreClient {
 		log.Fatal(err)
 	}
 
+	//Note 1:  the order of the fields is important and should match query order
+	//Note 2:  '-time' is the field we are sorting on must be the last field in the index
+	queryIndex := mgo.Index{
+		Key:        []string{"_groupId", "_active", "type", "-time"},
+		Background: true,
+	}
+	mongoSession.DB("").C(DEVICE_DATA_COLLECTION).EnsureIndex(queryIndex)
+
+	//As above but includes uploadId for restriction of data returned
+	queryUploadIdIndex := mgo.Index{
+		Key:        []string{"_groupId", "_active", "type", "uploadId", "-time"},
+		Background: true,
+	}
+	mongoSession.DB("").C(DEVICE_DATA_COLLECTION).EnsureIndex(queryUploadIdIndex)
+
```

Comment:
@jh-bate Have you gotten any performance results back from adding these indices on your development box?  Or will we try this on prod or staging?